integration/offline_test.go

```diff
@@ -87,5 +87,48 @@ func testOffline(t *testing.T, context spec.G, it spec.S) {
 			Expect(err).NotTo(HaveOccurred())
 			Expect(string(content)).To(ContainSubstring("Hello, World!"))
 		})
+
+		context("with custom vendor dir", func() {
```

Comment:
I went for an integration test because this introduces a "user-visible" feature. I wouldn't expect something this granular as an integration test at the language family (or samples/smoke test) level but it seems pretty lightweight here.

I'll pull it out if that's the desire.

Reply:
Understood, and makes sense. I'd probably pull it out because I think the unit tests provide sufficient confidence and are much faster to run than integration tests.

I know our integration tests are fairly lightweight and quick; perhaps I'm scarred by integration-test bloat that I've seen on other projects.

Reply:
@robdimsdale Test has been removed and the commit history rebased from main